octodns_gcore/__init__.py

```diff
@@ -580,6 +580,33 @@ def _apply(self, plan):
             class_name = change.__class__.__name__
             getattr(self, f"_apply_{class_name.lower()}")(change)
 
+    @staticmethod
+    def _validate_supported_country_code(record, geos, index=0):
+        filtered_geos = [
+            g for g in geos if g.startswith("NA-US-") or g.startswith("NA-CA-")
+        ]
+        if filtered_geos:
+            # We don't support provinces, we'll have to invalidate this rule
+            msg = f"NA-US- and NA-CA-* not supported for {record.fqdn}"
+            reason = f'rule {index} unsupported province code "{filtered_geos}"'
+            raise ValidationError(f"{msg}", [reason])
```

Comment:
There's a bit of info in the docs for `_process_desired_zone` about the preferred behavior for situations where a provider doesn't support a particular bit of functionality, https://github.com/octodns/octodns/blob/c9a57bf0a38b1affccf0c27738f7b76ba8416a70/octodns/provider/base.py#L49-L51.

Here that'd be calling 
Generally the preferred behavior is to call `supports_warn_or_except`. That allows users to specify whether they want to just get warnings about the problem and how it was fixed or (by default) if they want errors thrown.

For a simple example of how that works see:
https://github.com/octodns/octodns/blob/c9a57bf0a38b1affccf0c27738f7b76ba8416a70/octodns/provider/base.py#L58-L59

There's a more involved check & modification happening just below that around pool status, https://github.com/octodns/octodns/blob/c9a57bf0a38b1affccf0c27738f7b76ba8416a70/octodns/provider/base.py#L60 that might be closer to what's needed here. 

Main reason for this behavior is that someone may be using multiple providers and one of them may support the geo. If they are OK with the mismatch and prefer to use it when possible they an enable warnings and get continue with things as configured.

Reply:
Soon as I clicked the button I remembered Route53 actually filters out unsupported geos:

https://github.com/octodns/octodns-route53/blob/a5ec6ab9ec0b9b3e025837f8e3a0d1b7fb7c4d4c/octodns_route53/provider.py#L1142-L1179

Reply:
Thanks, I fixed it.